tests/test_wps_generate_prsn.py

```diff
@@ -1,137 +1,150 @@
 import pytest
+import re
 
 from pywps import Service
 from pywps.tests import assert_response_success
 
 from .common import TESTDATA, run_wps_process
 from thunderbird.processes.wps_generate_prsn import GeneratePrsn
 
+local_test_files = TESTDATA["test_local_nc"]
+# pr, tasmin and tasmax files to make vaild sets of test_data
+local_pr_files = [f for f in local_test_files if "pr" in f]
+local_tasmin_files = [f for f in local_test_files if "tasmin" in f]
+local_tasmax_files = [f for f in local_test_files if "tasmax" in f]
 
-def build_params(kwargs):
-    if (
-        "chunk_size" in kwargs.keys() and "output_file" in kwargs.keys()
-    ):  # Not using default values
+# creating sets of test_data with the matching files
+netcdf_sets = [
+    (pr, tasmin, tasmax)
+    for pr in local_pr_files
+    for tasmin in local_tasmin_files
+    for tasmax in local_tasmax_files
+    if re.sub("pr", "", pr) == re.sub("tasmin", "", tasmin)
+    and re.sub("pr", "", pr) == re.sub("tasmax", "", tasmax)
+]
+
+opendap_set = ["", "", ""]
+for od in TESTDATA["test_opendaps"]:
+    if re.search("\S*/pr_\S+.nc$", od):
+        opendap_set[0] = od
+    elif re.search("\S*/tasmin_\S+.nc$", od):
+        opendap_set[1] = od
+    elif re.search("\S*/tasmax_\S+.nc$", od):
+        opendap_set[2] = od
+
+opendap_set = [tuple(opendap_set)]
```

Comment:
Yes, a list containing a single tuple is my intention. There is only one set of opendap files that can be run in wps_generate_prsn.

Reply:
Ok, that isn't made clear in this code snippet. Plus if we were to add other files later on it could break this. Consider making it more specific or a more robust method.